modules/importer/src/runner/cwe/mod.rs

```diff
@@ -0,0 +1,59 @@
+mod walker;
+
+use crate::model::CweImporter;
+use crate::runner::{
+    context::RunContext,
+    cwe::walker::CweWalker,
+    report::{ReportBuilder, ScannerError},
+    RunOutput,
+};
+use std::sync::Arc;
+use tokio::sync::Mutex;
+use tracing::instrument;
+use trustify_module_ingestor::{graph::Graph, service::IngestorService};
+
+impl super::ImportRunner {
+    #[instrument(skip(self), ret)]
+    pub async fn run_once_cwe_catalog(
+        &self,
+        context: impl RunContext + 'static,
+        cwe_catalog: CweImporter,
+        continuation: serde_json::Value,
+    ) -> Result<RunOutput, ScannerError> {
+        let ingestor = IngestorService::new(Graph::new(self.db.clone()), self.storage.clone());
+
+        let report = Arc::new(Mutex::new(ReportBuilder::new()));
+        let continuation = serde_json::from_value(continuation).unwrap_or_default();
+
+        // no working-dir required
+
+        // one file, no progress to care about.
+
+        // run the walker
+
+        let walker = CweWalker::new(cwe_catalog.source.clone(), ingestor, report.clone())
+            .continuation(continuation);
+
+        match walker.run().await {
```

Comment:
note to self: but I've seen the same pattern a bunch of times now. Might be worth pulling that out into a common place.